std/math.d

```diff
@@ -7359,16 +7359,22 @@ T nextafter(T)(const T x, const T y) @safe pure nothrow @nogc
  *      $(TR $(TD x $(LT)= y) $(TD +0.0))
  *      )
  */
-real fdim(real x, real y) @safe pure nothrow @nogc { return (x > y) ? x - y : +0.0; }
+real fdim(real x, real y) @safe pure nothrow @nogc
+{
+    if (isNaN(x) || isNaN(y))
+        return real.nan;
+    return (x > y) ? x - y : +0.0;
+}
```

Comment:
You should be able to make this branchless by converting this to `return (x < y) ? +0.0 : x - y;`

see https://run.dlang.io/is/G0Ka0Z (LDC) DMD does about the same.

Reply:
But return (x < y) ? +0.0 : x - y; is not checking for NaN. Should I do it like return (isNaN(x) || isNan(y)) ? real.nan : ((x < y) ? +0.0 : x - y); ?

Reply:
Ah, but it is! 

https://run.dlang.io/is/o3siXK

The reasons is that `NaN ? a : b` is always `b`. by inverting the condition and swapping `a` and `b`, you get for any NaN `x` or `y`,`NaN ? +0.0 : x-y ` ->  `x-y`, and since one of them is NaN so is `x-y`.

Reply:
> `NaN ? a : b` is always `b`

Just to avoid confusion, you mean `x < y` is false when `x` and/or `y` are NaN.

NaN's boolean value is actually `true`, so `NaN ? a : b` is actually `a`.

Reply:
>NaN's boolean value is actually `true`

Thanks I forgot about that, how could I forget Sociomantic's $9T bug?

Reply:
Didn't knew that! Very interesting. Thank you. I will change the commit asap.

Reply:
> NaN's boolean value is actually true, so NaN ? a : b is actually a.

My bad, this is actually `NaN  > 0.0 ? a : b` and `NaN > 0.0` is always false. So I was right the first time :)

Reply:
Ping @ibuclaw